packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts

```diff
@@ -98,14 +98,22 @@ export class ChainAdapter
       const account = await this.getAccount(from)
 
       if (sendMax) {
-        tx.value = bnOrZero(account.balance).minus(gas).toString()
+        try {
+          const val = bnOrZero(account.balance).minus(gas)
+          if (!isFinite(val.toNumber()) || val.toNumber() < 0) {
+            throw new Error('CosmosChainAdapter: transaction value is invalid')
+          }
+          tx.value = String(val)
+        } catch (error) {
+          return ErrorHandler(error)
+        }
       }
 
       const utx: CosmosTx = {
         fee: {
           amount: [
             {
-              amount: bnOrZero(fee).toString(),
+              amount: String(bnOrZero(fee)),
```

Comment:
shouldn't we be calling .toString() with some precision stuff instead?

Reply:
The precision is handled on the frontend I believe, so I don't think we need to specify that. The `String` vs `toString` can go either way for me.